lib/indexer.rb

```diff
@@ -0,0 +1,151 @@
+module Argo
+  class SimpleLogJob
+    def initialize(log_val)
+      @log_val = log_val
+    end
+
+    def perform
+      sleep(rand 10)
+
+      log_filename = 'tmp/simple_job.log'
+      File.open(log_filename, 'a') { |log|
+        log.puts "#{@log_val}"
+      }
+    end
+  end
+
+  # TODO: DRY up the repetition btwn here and dor_controller
+  class Indexer
+    @@index_logger = nil
+    def self.index_logger
+      return @@index_logger if @@index_logger
+
+      @@index_logger ||= Logger.new("#{Rails.root}/log/indexer.log", 10, 3240000)
```

Comment:
I'm a bit :-1: on turning the rule off. I think there a lot of cases where a class instance variables aren't appropriate and having this rule may help raise those cases.